Bigtable/src/BigtableClient.php

```diff
@@ -0,0 +1,205 @@
+     * ```
+     *
+     * @param string $instanceId The instance ID
+     * @param array $instance [optional] {
+     *     Configuration options
+     *
+     *     @type string $displayName **Defaults to** the value of $instanceId.
+     *     @type int $instanceType Possible values are represented by the following constants:
+     *           `Google\Cloud\Bigtable\Instance::INSTANCE_TYPE_PRODUCTION` and
+     *           `Google\Cloud\Bigtable\Instance::INSTANCE_TYPE_DEVELOPMENT`.
+     *           **Defaults to** using `Google\Cloud\Bigtable\Instance::INSTANCE_TYPE_UNSPECIFIED`.
+     *     @type array $labels as key/value pair ['foo' => 'bar']. For more information, see
+     *           [Using labels to organize Google Cloud Platform resources]
+     *           (https://cloudplatform.googleblog.com/2015/10/using-labels-to-organize-Google-Cloud-Platform-resources.html).
+     * }
+     *
+     * @return Instance
+     */
+    public function instance($instanceId, array $instance = [])
+    {
+        return new Instance(
+            $this->connection,
+            $this->lroConnection,
+            $this->lroCallables,
+            $this->projectId,
+            $instanceId,
+            $instance
+        );
+    }
+
+    /**
+     * Helper method to create cluster metadata array for creating instances.
+     *
+     * Example:
+     * ```
+     * $cluster = $bigtable->buildClusterMetadata('my-cluster', 'us-east1-b');
+     * ```
+     * @param string $clusterId The cluster ID
+     *        e.g., just `cluster-id` rather than `projects/project-id/instances/instance-id/clusters/cluster-id`.
+     * @param string $locationId The location ID
+     *        e.g., just `us-east1-b` rather than `projects/project-id/locations/us-east1-b`.
+     * @param int $storageType The storage media type for persisting Bigtable data. Possible values include
+     *        `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_SSD`,
+     *        `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_HDD` and
+     *        `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_UNSPECIFIED`.
+     *        **Defaults to** `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_UNSPECIFIED`.
+     * @param int $serveNodes The number of nodes allocated to this cluster.
+     *        More nodes enable higher throughput and more consistent performance.
+     * }
+     *
+     * @return array
+     *
+     * @throws \InvalidArgumentException if invalid argument
+     */
+    public function buildClusterMetadata(
+        $clusterId,
+        $locationId,
+        $storageType = Instance::STORAGE_TYPE_UNSPECIFIED,
+        $serveNodes = null
+    ) {
+        $metaData = [];
+        $metaData['clusterId'] = $clusterId;
+        $metaData['locationId'] = $locationId;
+        $storageTypes = [
+            Instance::STORAGE_TYPE_UNSPECIFIED,
+            Instance::STORAGE_TYPE_SSD,
+            Instance::STORAGE_TYPE_HDD
+        ];
+        if (!in_array($storageType, $storageTypes)) {
```

Comment:
The previous discussion about this landed with us opting to let the server validate this value. We can remove this if, throw and the `$storageTypes` var above.